c_src/ep_encoder.c

```diff
@@ -704,6 +761,23 @@ pack_field(ErlNifEnv *env, ERL_NIF_TERM term, ep_enc_t *enc, ep_field_t *field)
     pack_tag(env, field->fnum, enc);
     sentinel = enc->p;
 
+    if (field->ebin == TRUE) {
+        switch (field->type) {
+        // currently, we only support int64.
+        case field_int64:
+            *(enc->sentinel) |= WIRE_TYPE_VARINT;
+            check_ret(ret, pack_bin_as_int64(env, term, enc));
```

Comment:
again I think we should just put this code in the right case below

Reply:
fixed.